src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java

```diff
@@ -233,5 +237,60 @@ public void moveToFolder(@Connection MailboxConnection connection,
     }
   }
 
+  /**
+   * Counts the emails in the {@code mailboxFolder}.
+   * <p>
+   *
+   * @param connection          The corresponding {@link MailboxConnection} instance.
+   * @param mailboxFolder       Mailbox folder where the emails are.
+   * @param countFilter         Count only messages specified with this option.
+   */
+  @Summary("Get the total amount of messages in a specified mailbox folder")
+  @DisplayName("Count messages - IMAP")
+  @Throws(EmailMarkingErrorTypeProvider.class)
```

Comment:
Is this the right error provider? Shouldn’t you have an EmailCountErrorTypeProvider?